src/Microsoft.OData.Core/Json/JsonValueUtils.cs

```diff
@@ -314,7 +314,7 @@ internal static void WriteValue(TextWriter writer, sbyte value)
         /// <param name="stringEscapeOption">The string escape option.</param>
         /// <param name="buffer">Char buffer to use for streaming data.</param>
         /// <param name="arrayPool">Array pool for renting a buffer.</param>
-        internal static void WriteValue(TextWriter writer, string value, ODataStringEscapeOption stringEscapeOption, ref char[] buffer, ICharArrayPool arrayPool = null)
+        internal static void WriteValue(TextWriter writer, string value, ODataStringEscapeOption stringEscapeOption, Ref<char[]> buffer, ICharArrayPool arrayPool = null)

```

Comment:
Does the use Ref<char[]> affect memory allocations on the sync writer?

Reply:
@habbes Passing around a `char` array reference in the synchronous workflow to support use of `char` array pool and a char array wrapped in an object in the asynchronous workflow started degenerating into a code smell so I opted for one that works across both workflows.
It required me to maintain both a `char[] buffer` and a `Ref<char[]> buffer` as member variables of the `JsonWriter` class, both renting  char arrays from the same pool, and returning to the same pool. The code felt brittle and likely to hide subtle bugs.
So the answer is yes, with this change both use the wrapped buffer.